### DISCUSSION TITLE
Performance difference between CLI and python script

### DISCUSSION THREAD

**Discussion by rodnvs** [nomadkaraoke/python-audio-separator#88] [General]
Hi @beveradb! I did a performance test for a 10min video, comparing the command line and the python script, and found a substantial difference in execution time. 
I only used the cpu for the test, because I don't have a gpu installed, and I ran the commands inside the vscode.

CLI - 26m46s
Python - 59m35s

Could you tell me what could be causing this?

### Command Line
`audio-separator video.mp4 --model_filename Kim_Vocal_2.onnx --output_format wav --single_stem Vocals`

### Python
```
from audio_separator.separator import Separator

# Initialize the Separator class (with optional configuration properties, below)
separator = Separator(
    output_single_stem='Vocals',
    output_format='WAV',
    sample_rate=44100,
    mdx_params={"hop_length": 1024, "segment_size": 256, "overlap": 0.25, "batch_size": 1, "enable_denoise": True}
)

# Load a machine learning model (if unspecified, defaults to 'model_mel_band_roformer_ep_3005_sdr_11.4360.ckpt')
separator.load_model(model_filename='Kim_Vocal_2.onnx')

# Perform the separation on specific audio files without reloading the model
output_files = separator.separate('video.mp4')

print(f"Separation complete! Output file(s): {' '.join(output_files)}")
```

**Comment by beveradb** [comment]
Hey @rodnvs , good question!

Can you try with identical config first? e.g. specify the same `mdx_params` you've set in your code on the CLI?
 
That would be my first guess, as you've enabled denoise in your code but not on the CLI.

You can see the default CLI params here: https://github.com/karaokenerds/python-audio-separator/blob/main/audio_separator/utils/cli.py#L59

also if you enable debug logging, the log output deliberately includes messages printing all the params, to help get to the bottom of things like this; see e.g.
- https://github.com/karaokenerds/python-audio-separator/blob/main/audio_separator/separator/common_separator.py#L86
- https://github.com/karaokenerds/python-audio-separator/blob/main/audio_separator/separator/architectures/mdx_separator.py#L63

Cheers,
Andrew

**Comment by rodnvs** [comment]
Hello @beveradb, thank you for your help!
I thought denoise was enabled by default in CLI. My mistake.
I re-tested using GPU, same video file, and disabling denoise. Now the separation times are much much better.

Audio Separator Kim 2 Vocals only Python CUDA - 01m06s
-------------------------------------------------
Audio Separator Kim 2 Vocals only CLI CUDA - 01m18s
-------------------------------------------------

I had problems getting it to run using the gpu but I managed to solve it using this help (I'm using python venv):
https://github.com/clementw168/install-and-test-gpu

You rock!!!
Rod

**Comment by beveradb** [comment]
Nice one! Yeah getting everything aligned for CUDA is a pain.
That seems like a thorough guide, maybe you could raise a PR updating the `audio-separator` README with a section linking to it for folks struggling to get CUDA working?
